atom/browser/api/atom_api_app.cc

```diff
@@ -144,6 +185,25 @@ void App::OnFinishLaunching() {
   Emit("ready");
 }
 
+void App::OnSelectCertificate(
+    content::WebContents* web_contents,
+    net::SSLCertRequestInfo* cert_request_info,
+    scoped_ptr<content::ClientCertificateDelegate> delegate) {
+  bool prevent_default =
+      Emit("select-certificate",
+           api::WebContents::CreateFrom(isolate(), web_contents),
+           cert_request_info->host_and_port.ToString(),
+           cert_request_info->client_certs,
+           base::Bind(&OnClientCertificateSelected,
+                      isolate(),
+                      base::Passed(&delegate)));
+
+  // Default to first certificate from the platform store.
+  if (!prevent_default)
+    delegate->ContinueWithCertificate(
```

Comment:
The ownership of `delegate` has been transferred so `delegate` is NULL here. You can put `delegate` in a shared_ptr to work around this.